tools/commands-packages-query.js

```diff
@@ -0,0 +1,1159 @@
+  }
+  // We don't know when the release was published. Luckily, since there is no
+  // way to use the tool outside of a release, and we always change the tool
+  // between releases, it is a good bet that the release was published on the
+  // same day as the tool.
+  var toolPackage = versionRecord.tool.split('@');
+  var toolName = toolPackage[0];
+  var toolVersion = toolPackage[1];
+  var toolRecord = catalog.official.getVersion(toolName, toolVersion);
+  return new Date(toolRecord.published);
+};
+
+// Convert package version dependencies into a more helpful format for output in
+// 'meteor show'.
+var formatDependencies = function (versionDependencies) {
+  var dependencies = _.map(
+    // The dependency on 'meteor' was almost certainly added automatically, by
+    // Isobuild. Returning this to the user will only cause confusion.
+    _.omit(versionDependencies, "meteor"),
+    function (dep, depName) {
+      // We will only consider this a weak dependency if all of its references
+      // are marked as weak.
+      var weak = _.every(dep.references, function (ref) {
+        return !! ref.weak;
+      });
+      return {
+        name: depName,
+        constraint: dep.constraint,
+        weak: weak
+      };
+  });
+  return dependencies;
+};
+
+// Converts an object to an EJSON string with the right spacing.
+var convertToEJSON = function (data) {
+  var EJSON = isopackets.load('ejson').ejson.EJSON;
+  return EJSON.stringify(data, { indent: true }) + "\n";
+};
+
+// Takes in a string and pads it with whitespace to the length of the longest
+// possible date string.
+var padLongformDate = function (dateStr) {
+  var numSpaces = utils.maxDateLength - dateStr.length;
+  return dateStr + Array(numSpaces + 1).join(' ');
+};
+
+// In order to get access to local package data, we need to create a local
+// package catalog. The best way to do that is to create a temporary
+// ProjectContext and let it handle catalog initialization. When we do, we need
+// to make sure that it is aware of all the local packages that we might care
+// about.
+//
+// This function returns such a ProjectContext, and takes in the following
+// options:
+//  - appDir: If we are running in the context of an app, this will contain the
+//    root of the app. We want to make sure to grab the data from the app's
+//    local packages.
+//  - packageDir: If we are running in a package directory, this will contain
+//    the source root of that package. If we are running from inside a package,
+//    we want that package to show up in our results.
+var getTempContext = function (options) {
+  var projectContext;
+  // If we are running in an app, we will use it to create a
+  // (mostly immutable) projectContext.
+  if (options.appDir) {
+    projectContext = new projectContextModule.ProjectContext({
+      projectDir: options.appDir,
+      // Don't change the state of the app.
+      neverWriteProjectConstraintsFile: true
```

Comment:
Isn't it better to be safe than sorry, in this case? I am afraid of (someone) accidentally running a command on the project that might, under some circumstances, try to modify things, so setting all flags that make sense seems like a good idea as a matter of principle. Is that incorrect? (Once again, not trying to argue, happy to remove extra flags, but I want to be smart in the future, too)